mosviz/controls/slit_controller.py

```diff
@@ -0,0 +1,181 @@
+import numpy as np
+
+from glue.core import HubListener
+
+from regions import RectangleSkyRegion, RectanglePixelRegion, PixCoord
+
+from matplotlib.patches import Rectangle
+
+from astropy.coordinates import Angle, SkyCoord
+from astropy.wcs.utils import proj_plane_pixel_area
+from astropy import units as u
+
+from ..controls.slit_selection_ui import SlitSelectionUI
+
+
+class SlitController(HubListener):
+    """
+    Controller that constructs and stores
+    a rectangular slit.
+    """
+
+    def __init__(self, mosviz_viewer=None):
+        super(SlitController, self).__init__()
+        self.mosviz_viewer = mosviz_viewer
+
+        self._slit = None  # `region.RectangleSkyRegion` object
+        self._pix_slit = None  # `region.RectanglePixelRegion` object
+        self._patch = None  # `matplotlib.patches.Rectangle` patch
+
+    @property
+    def is_active(self):
+        return self._patch is not None
+
+    @property
+    def patch(self):
+        return self._patch
+
+    @property
+    def x(self):
+        """Center x position of slit"""
+        if self.is_active:
+            return self.patch.get_x() + self.patch.get_width() / 2.
+
+    @property
+    def y(self):
+        """Center y position of slit"""
+        if self.is_active:
+            return self.patch.get_y() + self.patch.get_height() / 2.
+
+    @property
+    def width(self):
+        if self.is_active:
+            return self.patch.get_width()
+
+    @property
+    def length(self):
+        if self.is_active:
+            return self.patch.get_height()
+
+    @property
+    def dx(self):
```

Comment:
For the dx and dy, is there a reason to not use property names of `width` and `length` just so it is super clear what they refer to?

Reply:
The legacy code uses `dx` and `dy` to refer to the axis in the plots. If we ever decide to use rotated slits the `width` will not match the `dx` of that slit.

Reply:
Ok, cool, that is what I thought...